src/ResourceManager/AzureBatch/Commands.Batch/Pools/GetBatchPoolNodeCountsCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.Batch.Models;
+using System.Management.Automation;
+using Constants = Microsoft.Azure.Commands.Batch.Utils.Constants;
+
+namespace Microsoft.Azure.Commands.Batch
+{
+    [Cmdlet(VerbsCommon.Get, Constants.AzureBatchPoolNodeCounts, DefaultParameterSetName = Constants.AzureBatchPoolNodeCounts), OutputType(typeof(PSPoolNodeCounts))]
+    public class GetBatchPoolNodeCountsCommand : BatchObjectModelCmdletBase
+    {
+        private const int defaultMaxCount = 10;
+
+        [Parameter(Position = 0, ParameterSetName = Constants.PoolIdParameterSet, Mandatory = false,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The id of the pool for which to get node counts.")]
+        [ValidateNotNullOrEmpty]
+        public string PoolId { get; set; }
+
+        [Parameter(Position = 0, ParameterSetName = Constants.ParentObjectParameterSet, 
+            ValueFromPipeline = true, Mandatory = false, HelpMessage = "The pool object for which to get node counts.")]
+        [ValidateNotNullOrEmpty]
+        public PSCloudPool Pool { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ODataFilterParameterSet)]
+        public int MaxCount { get; set; } = defaultMaxCount;
```

Comment:
@antoniowmsft we should add the `ValidateRange` attribute to this parameter to limit the values that the user can provide